src/hash_map.c

```diff
@@ -109,6 +109,63 @@ void			hm_destroy(void *_hm, void (*f)(void *))
 	free(hm);
 }
 
+static t_hm_node		*find_node_by_key(void *_hm, char *_key)
+{
+	t_hash_map	*hm;
+	t_hm_node	*node;
+	
+	hm = (t_hash_map *)_hm;
+	node = hm->first_node;
+	while(node)
+	{
+		if (ft_strcmp(_key, node->key) == 0)
+			break ;
+		node = node->next;
+	}
+	if (!node || ft_strcmp(_key, node->key) != 0)
+		return (NULL);
+	return (node);
+}
+
+static void			node_remove_from_list(t_hash_map *_hm, t_hm_node *_node)
+{
+	if (_node->prev)
+		_node->prev->next = _node->next;
+	else
+		_hm->first_node = _node->next;
+	if (_node->next)
+		_node->next->prev = _node->prev;
+	else
+		_hm->last_node = _node->prev;
+}
+
+void				node_destroy(t_hm_node *_to_destroy, void (*_ft_delete)(void*))
+{
+	free(_to_destroy->key);
+	_ft_delete(_to_destroy->value);
+	free(_to_destroy);
+}
+
+void				hm_remove(void *_hm, char *_key, void (*_ft_delete)(void*))
+{
+	t_hash_map	*hm;
+	t_hm_node	**to_remove;
+	
+	hm = (t_hash_map *)_hm;
+	to_remove = &hm->nodes[hash(hm->cap, _key)];
+	if (*to_remove)
+	{
+		if (ft_strcmp((*to_remove)->key, _key) != 0)
+		{
+			*to_remove = find_node_by_key(_hm, _key);
```

Comment:
add argument to find_node_by_key : node, and start iterating over node->next starting on that note